src/main/java/ch/njol/skript/expressions/ExprVehicle.java

```diff
@@ -15,172 +16,138 @@
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
 import ch.njol.skript.doc.Since;
-import ch.njol.skript.effects.Delay;
 import ch.njol.skript.entity.EntityData;
-import ch.njol.skript.expressions.base.SimplePropertyExpression;
-import ch.njol.util.coll.CollectionUtils;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
 
-import java.lang.invoke.MethodHandle;
-import java.lang.invoke.MethodHandles;
-import java.lang.invoke.MethodType;
+import org.bukkit.entity.Player;
 
-/**
- * @author Peter Güttinger
- */
 @Name("Vehicle")
-@Description({"The vehicle an entity is in, if any. This can actually be any entity, e.g. spider jockeys are skeletons that ride on a spider, so the spider is the 'vehicle' of the skeleton.",
-		"See also: <a href='#ExprPassenger'>passenger</a>"})
-@Examples({"vehicle of the player is a minecart"})
+@Description({
+	"The vehicle an entity is in, if any.",
+	"This can actually be any entity, e.g. spider jockeys are skeletons that ride on a spider, so the spider is the 'vehicle' of the skeleton.",
+	"See also: <a href='#ExprPassenger'>passenger</a>"
+})
+@Examples({
+	"set the vehicle of {game::players::*} to a saddled pig",
+	"give {game::players::*} a carrot on a stick",
+	"",
+	"on vehicle enter:",
+		"\tvehicle is a horse",
+		"\tadd 1 to {statistics::horseMounting::%uuid of player%}",
+})
 @Since("2.0")
-public class ExprVehicle extends SimplePropertyExpression<Entity, Entity> {
-
-	private static final boolean HAS_NEW_MOUNT_EVENTS = Skript.classExists("org.bukkit.event.entity.EntityMountEvent");
-
-	private static final boolean HAS_OLD_MOUNT_EVENTS;
-	@Nullable
-	private static final Class<?> OLD_MOUNT_EVENT_CLASS;
-	@Nullable
-	private static final MethodHandle OLD_GETMOUNT_HANDLE;
-	@Nullable
-	private static final Class<?> OLD_DISMOUNT_EVENT_CLASS;
-	@Nullable
-	private static final MethodHandle OLD_GETDISMOUNTED_HANDLE;
+public class ExprVehicle extends PropertyExpression<Entity, Entity> {
 
 	static {
-		register(ExprVehicle.class, Entity.class, "vehicle[s]", "entities");
+		if (Skript.classExists("org.bukkit.event.entity.EntityMountEvent"))
+			registerDefault(ExprVehicle.class, Entity.class, "vehicle[s]", "entities");
```

Comment:
```suggestion
		registerDefault(ExprVehicle.class, Entity.class, "vehicle[s]", "entities");
```
Not needed since this will now be intended for 2.13,